app/models/ta.rb

```diff
@@ -36,28 +36,25 @@ def percentage_grades_array(assignment)
     grades = []
 
     if assignment.assign_graders_to_criteria
-      criteria_data = self.criterion_ta_associations.where(assessment_id: assignment.id)
-                          .pluck(:criterion_type, :criterion_id)
-      out_of = criteria_data.sum do |criterion_type, criterion_id|
-        criterion_type.constantize.find(criterion_id).max_mark
+      criteria_data = self.criterion_ta_associations.where(assessment_id: assignment.id).pluck(:criterion_id)
```

Comment:
You can simplify this logic even further. Something like `self.criteria.where('criteria.assessment_id': assignment.id).sum(:max_mark)`.

Reply:
We can't change this logic because we use `criteria_data` later when finding mark data, so we still want to keep track of the ids. Rather we should probably just change the name to `criteria_ids`.